### PR TITLE
Life Science assets & system tests migration (AIP-47)

### PR DESCRIPTION
**Pull request by bkossakowska**
Life Science assets & system tests migration (AIP-47)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Needs rebase I am afraid :(

File: airflow/providers/google/provider.yaml [hunk 1]
```diff
@@ -984,6 +984,8 @@ extra-links:
   - airflow.providers.google.cloud.links.cloud_memorystore.MemcachedInstanceListLink
   - airflow.providers.google.cloud.links.cloud_memorystore.RedisInstanceDetailsLink
   - airflow.providers.google.cloud.links.cloud_memorystore.RedisInstanceListLink
+  - airflow.providers.google.cloud.links.life_sciences.LifesciencesLink
```
Comment (josh-fell) [inline airflow/providers/google/provider.yaml]:
```suggestion
  - airflow.providers.google.cloud.links.life_sciences.LifesciencesLink
```
Looks like a duplicate of sorts and the link is named `LifeSciencesLink` which has an entry on the next line.

File: airflow/providers/google/cloud/links/life_sciences.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    name = "Life Sciences"
+    key = "lifesciences_key"
+    format_str = LIFESCIENCES_LIST_LINK
+
+    @staticmethod
+    def persist(
+        context: "Context",
+        task_instance,
+        project_id: Optional[str],
```
Comment (josh-fell) [inline airflow/providers/google/cloud/links/life_sciences.py]: Similar to the [comment in #25466](https://github.com/apache/airflow/pull/25466#discussion_r943033290), based on the URL format `project_id` is required but the typing effectively says both a str and `None` are valid arg types. Perhaps the "proper" fix is to change the typing to `project_id: str` and add a check to ensure `project_id` is not `None` when `persist()` is called in the operator? WDYT?

Comment (bkossakowska) [reply airflow/providers/google/cloud/links/life_sciences.py]: yes, I have changed optional project_id to required

**Comment by potiuk** [comment]
Needs resolving conflicts @bkossakowska !

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool. sorry for some delays .... slowly catching up

**Comment by lwyszomi** [comment]
https://github.com/apache/airflow/issues/9941

**Pull request merged**